src/frontend-samples/explode-sample/ExplodeApp.tsx

```diff
@@ -0,0 +1,178 @@
+          vp.zoomToVolume(volume, options);
+          ExplodeTreeReference.onTreeDataUpdated.removeListener(awaitRangeLoaded);
+        }
+      };
+      const removeListener = ExplodeTreeReference.onTreeDataUpdated.addListener(awaitRangeLoaded);
+      ExplodeApp.cleanUpCallbacks.push(removeListener);  // This will insure the listener is removed before swapping samples.
+    } else {
+      vp.zoomToVolume(volume, options);
+    }
+  }
+
+  /** Updates the tile tree reference with the given data and signals the viewport that changes have been made. */
+  public static refSetData(vp: Viewport, name: string, ids: string[], explodeScaling: number) {
+    const provider = ExplodeProvider.getOrCreate(vp);
+    provider.setData(name, ids, explodeScaling);
+
+    // Necessary when the data changes.
+    provider.invalidate();
+    // Necessary when the explode scaling changes.
+    ExplodeApp.invalidateScene(vp);
+  }
+
+  /** Uses the Viewport API to signal that the scene needs to re-render. */
+  public static invalidateScene(vp: Viewport) {
+    vp.invalidateScene();
+  }
+
+  /** Enables an animator using the Viewport API. If the animator is undefined, any active animator will be removed. */
+  public static setAnimator(vp: Viewport, animator?: Animator) {
+    vp.setAnimator(animator);
+  }
+}
+
+/** This provider both hides the original graphics of the element and inserts the transformed graphics. */
+class ExplodeProvider implements TiledGraphicsProvider, FeatureOverrideProvider {
+  // These methods support the provider as an object.
+  /** Returns a provider associated with a given viewport. If one does not exist, it will be created. */
+  public static getOrCreate(vp: Viewport) {
+    let provider = vp.findFeatureOverrideProviderOfType(ExplodeProvider);
+    if (!provider)
+      provider = ExplodeProvider.createAndAdd(vp);
+    return provider;
+  }
+  /** Creates a provider associated with a given viewport. */
+  public static createAndAdd(vp: Viewport) {
+    const provider = new ExplodeProvider(vp);
+    provider.add(vp);
+    return provider;
+  }
+
+  /** Updates the TileTree with the elements and explode scaling. */
+  public setData(name: string, elementIds: string[], explodeScaling: number) {
+    this.explodeTileTreeRef.explodeScaling = explodeScaling;
+    this.explodeTileTreeRef.setExplodeObject(name, elementIds);
+  }
+  /** Adds provider from viewport */
+  public add(vp: Viewport) {
+    if (!vp.hasTiledGraphicsProvider(this)) {
+      vp.addTiledGraphicsProvider(this);
+      vp.addFeatureOverrideProvider(this);
+    }
+  }
+  /** Drops provider from viewport */
+  public drop() {
+    this.vp.dropFeatureOverrideProvider(this);
+    this.vp.dropTiledGraphicsProvider(this);
+  }
+
+  /** Signals to the viewport that this provider needs to be recalculated. */
+  public invalidate() {
+    this.vp.setFeatureOverrideProviderChanged();
+  }
+
+  // These methods support the functionality of the provider.
```

Comment:
// These methods support the functionality of the provider.

I suggest removing this comment